EXO-60596: migrate database configuration page to new documentation (#9)
exoplatform/exoplatform.github.io: docs/administration/database.md

>
> In this chapter:
>
> - `Creating databases` You need to create databases. The tables are created automatically in Platform first startup.
> - `Configuring eXo Platform` This section is a complete guide to configure eXo to use a RDBMS.
> - `Configuring Database in a docker install`  This section describes how to configure database in a docker container.
> - `Datasource JNDI name` If for any reason you want to change JNDI name of the datasources, follow this section.
> - `File storage` This section targets to describe the possible ways to store binary files in eXo Platform. It also makes comparison between them to allow administrators to choose the suitable file storage for their environment.
> - `Configuring eXo Chat database` This section introduces the parameters to configure eXo Chat MongoDB database.
> - `Frequently asked questions` This section especially targets to solve some possible issues.

## Creating databases

eXo Platform uses three datasources:

- **IDM**: For Identity Management service.
- **JCR**: For Java Content Repository service.
- **JPA**: To store entities mapped by Hibernate.

> Create 3 databases, one for IDM, one for JCR and one for JPA, or you can use the same database for all pf them. Leave the databases empty, the tables will be created automatically in Platform first startup.

Reviewer: ```suggestion
Create 3 databases, one for IDM, one for JCR and one for JPA, or you can use the same database for all of them. Leave the databases empty, the tables will be created automatically on Platform first startup.
```

     ```TEXT
        url="jdbc:mysql://localhost:3306/plf?autoReconnect=true&amp;characterEncoding=utf8"
     ```

2. Set the SQL Dialect if necessary. This step is not mandatory because the dialect is auto-detected in most cases. You only need to take care of it for some particular RDBMSs:
   - For JCR, only when you are using MySQL and database character set `utf8`, you need to edit `gatein/conf/exo.properties` file to have:

     ```properties
        exo.jcr.datasource.dialect=MySQL-UTF8
     ```

   - For IDM, eXo Platform detects automatically the dialect for RDBMSs listed [here](http://docs.jboss.org/hibernate/orm/4.1/manual/en-US/html_single/#configuration-optional-dialects). Only when your RDBMS is **not** in the list, for example *Postgres Plus Advanced Server 9.2*, you will need to edit `gatein/conf/exo.properties` file to have:

     ```properties
       hibernate.dialect=org.hibernate.dialect.PostgresPlusDialect
     ```

     Or for *Oracle Database 12c*:

     ```properties
        hibernate.dialect=org.hibernate.dialect.Oracle10gDialect
     ```

3. Download the JDBC driver for Java and install it to `$PLATFORM_TOMCAT_HOME/lib`.

::: tip

> Normally you can find out an appropriate driver for your JDK from your database vendor website. For example, for MySQL: <http://dev.mysql.com/downloads/connector/j/>, and for Oracle: <http://www.oracle.com/technetwork/database/features/jdbc/index-091264.html>.

Reviewer: should we add the possibility to install the related addon ?

Author: yes I think this will be helpful

Author: yes done

::: tip

The properties file (`exo.properties`) will not take **_portal** suffix as it is appended automatically by eXo, as detailed below.
:::

There is a constraint that the suffix of datasource JNDI names must be **_portal**. Take JCR as example, it uses the following property:

```properties
    exo.jcr.datasource.name=java:/comp/env/exo-jcr
```

to look up a datasource for the portal. Because the core of eXo Platform is designed for supporting multi-portal, there are theoretically different datasources for different portals. Consequently this property is treated as datasource name\'s prefix, and the portal name (knowing that it is \"portal\" by default) is appended to complete the name in JNDI lookup.

So if you change the JDNI names *exo-idm_portal* and **exo-jcr_portal**, you need to edit the following properties in `gatein/conf/exo.properties`:

```properties  
  # JNDI Name of the IDM datasource
  exo.idm.datasource.name=java:/comp/env/exo-idm
  ...
  # name of the datasource that will be used by eXo JCR
  exo.jcr.datasource.name=java:/comp/env/exo-jcr
```

You also need to edit `conf/Catalina/localhost/context.xml.default` file:

> 

Reviewer: I'm sure this part is pertinent.
What do you think ?

Author: @rdenarie this is for informative purpose and won't harm to put it there. I migrated the original content, and I think it s worth knowing as a side note

Reviewer: I dont want to make the documentation too complex to read and understand, and this part add complexity.
For us, we are knowing this and we understand it. For a newbie, it could be complex to understand, that's why I prefer to remove it

Author: Ok

Author: @rdenarie done
